samples/create-buffers-around-points/src/main/java/com/esri/arcgismaps/sample/createbuffersaroundpoints/components/CreateBuffersAroundPointsViewModel.kt

```diff
@@ -0,0 +1,291 @@
+                    style = SimpleFillSymbolStyle.Solid,
+                    color = Color.fromRgba(r = 255, g = 255, b = 0, a = 153),
+                    outline = bufferOutline
+                )
+                bufferGraphicsOverlay.renderer = SimpleRenderer(symbol = bufferFill)
+
+                // Configure tapped points overlay renderer (small red circles)
+                val tapSymbol = SimpleMarkerSymbol(
+                    style = SimpleMarkerSymbolStyle.Circle,
+                    color = Color.red,
+                    size = 10f
+                )
+                tappedPointsGraphicsOverlay.renderer = SimpleRenderer(symbol = tapSymbol)
+
+                //Set initial viewpoint of the map
+                arcGISMap.initialViewpoint = Viewpoint(boundingGeometry = boundaryPolygon as Geometry)
+                // Load the map; report any failure
+                arcGISMap.load().onFailure { messageDialogVM.showMessageDialog(it) }
+
+
+            } catch (e: Throwable) {
+                messageDialogVM.showMessageDialog(e)
+            }
+        }
+    }
+
+    /**
+     * Called by the MapView when the user taps the map. If the tap is within the valid boundary,
+     * request a buffer radius input from the UI. Otherwise update status to indicate out-of-bounds.
+     */
+    fun onMapTapped(mapPoint: Point) {
+        viewModelScope.launch {
+            val boundary = boundaryPolygon
+            if (boundary == null) {
+                _statusText.value = "Boundary not initialized"
+                return@launch
+            }
+
+            val contains = GeometryEngine.contains(boundary, mapPoint)
+            if (!contains) {
+                _statusText.value = "Tap within the boundary to add buffer."
+                // Inform the user via message dialog as well
+                messageDialogVM.showMessageDialog("Tap within the boundary to add buffer.")
+                return@launch
+            }
+
+            // Store the tapped point and request input dialog
+            lastTappedPoint = mapPoint
+            _isInputDialogVisible.value = true
+            _statusText.value = "Enter buffer radius (miles) for the tapped location"
+        }
+    }
+
+    /**
+     * Submit the radius (in miles) provided by the user. Converts miles to the map's linear units
+     * (the state plane spatial reference uses US feet) then creates the buffer and updates the overlays.
+     */
+    fun submitRadiusMiles(radiusMiles: Double) {
+        viewModelScope.launch {
+            val point = lastTappedPoint
+            if (point == null) {
+                messageDialogVM.showMessageDialog("Internal error: missing tapped point")
+                _isInputDialogVisible.value = false
+                return@launch
+            }
+
+            if (radiusMiles <= 0.0) {
+                messageDialogVM.showMessageDialog("Please enter a value greater than 0")
+                return@launch
+            }
```

Comment:
Let's check for both bounds: 

```suggestion
        if (radiusMiles <= 0.0 || radiusMiles >= 300) {
            return messageDialogVM.showMessageDialog("Please enter a value between 0 & 300.")
        }
```